framework/src/base/MeshGeneratorSystem.C

```diff
@@ -0,0 +1,496 @@
+    if (ancestor_list.size() != resolver.size() && _final_generator_name.empty())
+    {
+      // Need to remove duplicates and possibly perform a difference so we'll import out list
+      // into a set for these operations.
+      std::set<MeshGenerator *, MeshGenerator::Comparator> ancestors(ancestor_list.begin(),
+                                                                     ancestor_list.end());
+      // Get all of the items from the resolver so we can compare against the tree from the
+      // final generator we just pulled.
+      const auto & allValues = resolver.getSortedValues();
+      decltype(ancestors) all(allValues.begin(), allValues.end());
+
+      decltype(ancestors) ind_tree;
+      std::set_difference(all.begin(),
+                          all.end(),
+                          ancestors.begin(),
+                          ancestors.end(),
+                          std::inserter(ind_tree, ind_tree.end()));
+
+      std::ostringstream oss;
+      oss << "Your MeshGenerator tree contains multiple possible generator outputs :\n\""
+          << final_generators.back()->name()
+          << " and one or more of the following from an independent set: \"";
+      bool first = true;
+      for (const auto & gen : ind_tree)
+      {
+        if (!first)
+          oss << ", ";
+        else
+          first = false;
+
+        oss << gen->name();
+      }
+      oss << "\"\n\nThis may be due to a missing dependency or may be intentional. Please "
+             "select the final MeshGenerator in\nthe [Mesh] block with the \"final_generator\" "
+             "parameter or add additional dependencies to remove the ambiguity.";
+      mooseError(oss.str());
+    }
+
+    _final_generator_name = final_generators.back()->name();
+  }
+  else
+    mooseAssert(hasMeshGenerator(_final_generator_name), "Missing the preset final generator");
+}
+
+void
+MeshGeneratorSystem::executeMeshGenerators()
+{
+  libmesh_parallel_only(comm());
+
+  // we do not need to do this when there are no mesh generators
+  if (_mesh_generators.empty())
+    return;
+
+  // Order the generators
+  createMeshGeneratorOrder();
+
+  // Grab the outputs from the final generator so MeshGeneratorMesh can pick them up
+  _final_generated_meshes.emplace_back(&getMeshGeneratorOutput(_final_generator_name));
+
+  // Need to grab two if we're going to be making a displaced mesh
+  if (_app.actionWarehouse().displacedMesh())
+    _final_generated_meshes.emplace_back(&getMeshGeneratorOutput(_final_generator_name));
+
+  // Run the MeshGenerators in the proper order
+  for (const auto & generator_set : _ordered_mesh_generators)
+  {
+    for (const auto & generator : generator_set)
+    {
+      const auto & name = generator->name();
+
+      auto current_mesh = generator->generateInternal();
+
+      // Now we need to possibly give this mesh to downstream generators
+      auto & outputs = _mesh_generator_outputs[name];
```

Comment:
where is the list being created? cant seem to find it

Reply:
https://github.com/loganharbour/moose/blob/38f9ecf998eecc9199f203e0e7e0031b7343f755/framework/src/base/MeshGeneratorSystem.C#L433 this is not a new API